qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py

```diff
@@ -296,23 +296,19 @@ def _autoscale_clim(self):
         """Update stored colorbar limits
         The new limits are found from the colobar data """
         data = self.colorbar.mappable.get_array_clipped_to_bounds()
-
-        # If any dimension is zero then we have no data in the display area
-        if any(map(lambda dim: dim == 0, data.shape)):
-            self.cmin_value = 0.
-            self.cmax_value = 0.
-            self.update_clim_text()
-            return
-
         norm = NORM_OPTS[self.norm.currentIndex()]
         try:
             try:
                 masked_data = data[~data.mask]
                 # use the smallest positive value as vmin when using log scale,
                 # matplotlib will take care of the data skipping part.
                 masked_data = masked_data[masked_data > 0] if norm == "Log" else masked_data
-                self.cmin_value = masked_data.min()
-                self.cmax_value = masked_data.max()
+
+                # If any dimension is zero then we have no data in the display area
+                data_is_empty = any(map(lambda dim: dim == 0, masked_data.shape))
```

Comment:
This works nicely now - I note that this doesn't cause an issue for log colorscale as when you get limits set to (0,0) then it switches the scale to linear.